src/update_worker.rs

```diff
@@ -62,71 +69,100 @@ impl UpdateWorker {
             client: Client::new(),
             queries,
             verbose,
+            behav,
         })
     }
 
-    async fn read_current_state(&self, subject: &str) -> reqwest::Result<DbState> {
-        let state = self
+    async fn read_current_state(&self, subject: &str) -> reqwest::Result<Option<DbState>> {
+        let resp = self
             .client
             .get(self.query_endpoint.clone())
             .header(header::ACCEPT, "application/n-triples")
             .query(&[("query", Self::make_verify_query(subject))])
             .send()
-            .await?
-            .text()
-            .await?;
+            .await;
 
-        let mut ret: DbState = state.lines().map(ToOwned::to_owned).collect();
-        ret.sort();
+        match resp {
+            Ok(resp) => {
+                let resp = resp.error_for_status()?;
+                match resp.text().await {
+                    Ok(state) => {
+                        let mut ret: DbState = state.lines().map(ToOwned::to_owned).collect();
+                        ret.sort();
 
-        Ok(ret)
+                        Ok(Some(ret))
+                    },
+                    Err(_) if self.behav == WorkerBehaviour::IgnoreConnectionError => Ok(None),
+                    Err(e) => Err(e),
+                }
+            },
+            Err(_) if self.behav == WorkerBehaviour::IgnoreConnectionError => Ok(None),
+            Err(e) => Err(e),
+        }
     }
 
-    async fn issue_update(&self, update: &str) -> reqwest::Result<()> {
-        self.client
+    async fn issue_update(&self, update: &str) -> reqwest::Result<Option<()>> {
+        let resp = self
+            .client
             .post(self.update_endpoint.clone())
             .header(header::CONTENT_TYPE, "application/sparql-update")
             .body(update.to_owned())
             .send()
-            .await?
-            .error_for_status()?;
-
-        Ok(())
+            .await;
+
+        match resp {
+            Ok(resp) => {
+                resp.error_for_status()?;
+                Ok(Some(()))
+            },
+            Err(_) if self.behav == WorkerBehaviour::IgnoreConnectionError => Ok(None),
+            Err(e) => Err(e),
+        }
     }
 
     pub async fn execute(&self) -> Result<(), WorkerError> {
         for (id, (subject, update, expected_state)) in self.queries.iter().enumerate() {
-            self.issue_update(update)
-                .await
-                .map_err(|err| WorkerError::UpdateFailed {
-                    update_id: id,
-                    err,
-                    verbose_info: if self.verbose {
-                        Some(UpdateFailedVerboseInfo { query: update.clone() })
-                    } else {
-                        None
+            loop {
```

Comment:
why do we need a loop now? same goes for the `read_current_state` below

Reply:
Because in the new mode, queries might have a connection error. This however is not considered an error in this mode. Therefore we just repeat the query until we get a successful connection.

Note that invalid data (unrelated to connection issues) is *not* ignored.

Reply:
Yes, we still want the update worker to execute every single update and verify the response and not just skip some while the server is down.